cli/azd/cmd/auth_login.go

```diff
@@ -139,6 +141,16 @@ func (lf *loginFlags) Bind(local *pflag.FlagSet, global *internal.GlobalCommandO
 		cClientCertificateFlagName,
 		"",
 		"The path to the client certificate for the service principal to authenticate with.")
+	local.StringVar(
+		&lf.serviceConnectionID,
+		"service-connection-id",
```

Comment:
Ideally, I would like to hide all the implementation details from the connection because it is 100% unique to Azure Devops.
Azd currently has the flag `--federated-credential-provider` to hide all the complexity and implementation of how to do OICD in github actions. That means, when you want to log in azd in in GitHub actions, you do:

```
run: |
          azd auth login `
            --client-id "$Env:AZURE_CLIENT_ID" `
            --federated-credential-provider "github" `
            --tenant-id "$Env:AZURE_TENANT_ID"
```

The implementation about how to get the system token and any other tokens is inside azd's implementation.

So, I would like to just have a new provider `azdo`, which I can use in a Azdo task like:

```
run: |
          azd auth login `
            --client-id "$Env:AZURE_CLIENT_ID" `
            --federated-credential-provider "azdo" `
            --tenant-id "$Env:AZURE_TENANT_ID"
```

Azd would internally know what ENV VARS to looks for to get the system access token and the connection-id.
Azd can adapte itself to multiple scenarios. For example, running `azd login` from the AzureCLI task.

Reply:
I can look into this but I think the problem is that without the connection id or service connection name we are not going to be able to make this work.  We could consider something like `--federated-credential-provider azdo:<service-connection-name>` so it would be something like `--federated-credential-provider azdo:azure-sdk-tests` for us.  Maybe we can make that work?

Like you, I'd love to be able to do this.

Reply:
Once we're done with investigation, I'd love to be able to make sense of whether `azd` should just be an AzDo task to support this scenario better (my hunch was always that it is indeed necessary, without moving mountains in AzDo).

If we are able to deliver an up-to-par experience, I'd love for that to be captured in #4341

Reply:
Made some progress here, we now support using `azure-pipelines` as an argument to `--federated-credential-provider`.  We now sniff the environment variables that the AzCli and AzurePowerShell tasks set, or if you want to use a different task, you can explicitly set `AZURESUBSCRIPTION_SERVICE_CONNECTION_ID`.

We may want to end up writing our own wrapper task at some point.

Reply:
I started an azd task for azdo some time ago. Basically copying what the AzureCLI task does here: https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks/AzureCLIV2

I stopped myself and started wondering if we want to update `AzureCLIV2` to check if azd is installed and login azd as well as az. It would be one less task to install for customers.